transaction-volume-anomaly-detection-template/src/TimeSeriesDeviationTracker.js

```diff
@@ -0,0 +1,122 @@
+/*
+  Per block we have x amount of transactions that will get added to the obj
+  So RollinMath should add the transactions per block for the blockSize we want to look up to -> for example 5 blocks
+  So we add the number to the bucket and we dont need to generate extra buckets per block, instead we need to have 1 rolling math object for the range and it will
+  update accordingly when a new block is added 
+*/
+const { default: BigNumber } = require("bignumber.js");
+const RollingMath = require("rolling-math");
+
+class TimeSeriesAnalysis {
+  constructor(contractAddress, blockSize) {
+    this.contract = contractAddress;
+    this.blockSize = blockSize;
+    this.bucket = new RollingMath(blockSize);
+    this.stdForAll = [];
+    this.timeline = 5; //the time in minutes we need to track to check for unusual counts
+    this.currentBlock = 0;
+    this.lastBlock = 0;
+    this.total = 0; //total transactions that complete our requirements
+  }
+
+  //Here we change the active block number so we know if we need to keep track of txs
+  updateBlock(blockNumber) {
+    if (this.currentBlock != blockNumber) {
+      this.lastBlock = this.currentBlock;
+      this.currentBlock = blockNumber;
+    }
+  }
+
+  updateCurrentBlock(blockNumber) {
+    this.currentBlock = blockNumber;
+  }
+
+  //Logic for adding a transaction to the bucket
+  AddTransaction(tx) {
+    if (tx.block.number != this.currentBlock && this.currentBlock != 0) {
+      this.updateBlock(tx.block.number);
+      this.AddToBucket(this.total);
+
+      this.total++;
+    } else if (tx.block.number == this.currentBlock) {
+      this.total++;
+    } else if (this.currentBlock == 0) {
+      this.total++;
+      this.updateCurrentBlock(tx.block.number);
+    }
+  }
+
+  AddToBucket(data) {
+    this.bucket.addElement(new BigNumber(data));
+    this.stdForAll.push(this.bucket.getStandardDeviation().toNumber());
+    this.total = 0;
+  }
+
+  GetStdForLatestBucket() {
+    return this.bucket.getStandardDeviation().toNumber();
+  }
+
+  GetTotalForLastBucket() {
+    return this.bucket.getSum().toNumber();
+  }
+
+  GetBaselineForLastBucket() {
+    return this.bucket.getAverage().toNumber();
+  }
+
+  //We need the SMA to calculate the EMA for all transactions
+  SimpleMovingAverage(prices, window, n = Infinity) {
```

Comment:
any reason why you picked this vs the time series anomaly detection approach?

Reply:
i may be a bit confused given that the unit test refers to timeseries

Reply:
This is the template which is still not yet updated